odl/tomo/data/uncompr_bin.py

```diff
@@ -0,0 +1,472 @@
+    |'size_bytes'  |int      |Size of the current element in bytes      |
+    +--------------+---------+------------------------------------------+
+    |'dtype'       |type     |Data type of the current element as       |
+    |              |         |defined by Numpy                          |
+    +--------------+---------+------------------------------------------+
+    |'description' |string   |Description of the element (optional)     |
+    +--------------+---------+------------------------------------------+
+    |'dshape'      |tuple    |For multi-elements: number of elements per|
+    |              |         |dimension. Optional for single elements.  |
+    +--------------+---------+------------------------------------------+
+
+    Parameters
+    ----------
+    field_list : sequence of dicts
+        Dictionaries describing the field elements, as returned by, e.g.,
+        `mrc_spec_fields`.
+    keys : dict
+        Dictionary with the following entries for the column headers in
+        the specification table:
+
+            - ``'id'``
+            - ``'byte_range'``
+            - ``'dtype'``
+            - ``'name'``
+            - ``'description'``
+
+    dtype_map : dict
+        Mapping from the data type specifiers in the specification table
+        to NumPy data types.
+
+    Returns
+    -------
+    standardized_fields : tuple of dicts
+        The standardized fields according to the above table.
+    """
+    # Parse the fields and represent them in a unfied way
+    conv_list = []
+    for row, field in enumerate(field_list):
+        new_field = {}
+
+        # Name and description: lowercase name, copy description
+        new_field['name'] = field[keys['name']].lower()
+        new_field['description'] = field[keys['description']]
+
+        # Get offset from ID range
+        num_range = field[keys['id']].split('-')
+        nstart = int(num_range[0])
+        nend = int(num_range[-1])  # 0 for range of type 3-
+
+        # Get byte range and set start
+        byte_range = field[keys['byte_range']].split('-')
+        byte_start = int(byte_range[0]) - 1
+        byte_end = int(byte_range[-1]) - 1
+        new_field['offset_bytes'] = byte_start
+
+        # Data type: transform to Numpy format and get shape from its
+        # itemsize and the byte range
+        dtype = dtype_map[field[keys['dtype']]]
+        byte_size = byte_end - byte_start + 1
+
+        if hasattr(dtype, 'itemsize') and byte_size % dtype.itemsize:
+            raise ValueError(
+                'in row {}: byte range {} not a multiple of itemsize {} '
+                'of the data type {}.'
+                ''.format(row + 1, field[keys['byte_range']], dtype.itemsize,
+                          field[keys['dtype']]))
+
+        new_field['dtype'] = dtype
+        new_field['size_bytes'] = byte_size
+        # Assuming 1d arrangement of multiple elements
+        # TODO: find way to handle 2d fields
```

Comment:
A suggestion would be to use a data type `Float32(3,2)` if a certain shape is desired. The function should in any case explain how the table is supposed to look like.